portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACTests.macro

```diff
@@ -1,13 +1,30 @@
 definition {
 
 	macro deleteABTest {
-		ACNavigation.openItem(itemName = ${testName});
+		if (isSet(testName)) {
+			ACNavigation.openItem(itemName = ${testName});
+		}
 
-		Click(locator1 = "ACTests#AB_ELLIPSIS");
+		Button.clickDelete();
+
+		SelectWindow(locator1 = "name=undefined");
+
+		Click(locator1 = "ACSettings#CONFIRMATION_BUTTON");
 
-		Click(locator1 = "ACTests#DELETE_TEST");
+		Alert.viewSuccessMessage();
+	}
 
+	macro deleteABTestACSide {
 		Button.clickDelete();
+
+		Click(locator1 = "ACSettings#CONFIRMATION_BUTTON");
+	}
```

Comment:
Question about these two macros. It looks like the functions in the macros are switched. Since if we are using `deleteABTestACSide` shouldn't it need the testName variable and the `SelectWindow` function? Or maybe just need to rename the `deleteABTestACSide` to `deleteABTestDXPSide`. Correct me if I'm wrong.

Reply:
Hey Josh, thanks for reviewing it (I know it's a lot).
I believe the first macro `ACTests.deleteABTest` is used when we still have the AB Test on DXP, so when we click to delete the test in AC we will be redirected to DXP.

The second macro `ACTests.deleteABTestACSide` is for when we don't have the test in DXP anymore, like when we complete or terminate a test then create a new one, in this case the test will be only available to delete on AC side, and we will not be redirected to DXP in that case. I created this macro because we might need to use it in the future (since we don't know what other changes the AB test will bring).

In case you are wondering why I chose not to put the `ACNavigation.openItem(itemName = ${testName});` inside this macro that I created is because I don't think the aim of the macro is to click and access the test and then delete, I believe the goal of the macro is only to delete. I followed the separation of concern (SOC) pattern where a method should only focus on a single specific responsibility or task.

is this what you meant? I hope I explained well :slightly_smiling_face:

Reply:
Ohh got it! Thanks for the explanation. That makes sense to me. 
I was confused with `ACTests.deleteABTestACSide` since I thought it was the deletion that should redirect to DXP. Didn't realize it was for when the test no longer in DXP.
Great job! I should hopefully be finished soon with reviewing.

Reply:
Hey Josh, thank you! If you need me to change or explain something more let me know :slightly_smiling_face: